inst/tests/tests.Rraw

```diff
@@ -18292,3 +18292,41 @@ DT = data.table(A=1:3, key="A")
 test(2223.1, DT[.(4), nomatch=FALSE], data.table(A=integer(), key="A"))
 test(2223.2, DT[.(4), nomatch=NA_character_], data.table(A=4L, key="A"))
 
+# gshift, #5205
+options(datatable.optimize = 2L)
+esc = shift
+DT = data.table(x = sample(letters[1:5], 20, TRUE),
+                y = rep.int(1:2, 10),  # to test 2 grouping columns get rep'd properly
+                i = sample(c(-2L,0L,3L,NA), 20, TRUE),
+                d = sample(c(1.2,-3.4,5.6,NA), 20, TRUE),
+                s = sample(c("foo","bar",NA), 20, TRUE),
+                c = sample(c(0+3i,1,-1-1i,NA), 20, TRUE),
+                l = sample(c(TRUE, FALSE, NA), 20, TRUE),
+                r = as.raw(sample(1:5, 20, TRUE)),
+if (test_bit64) i64 = as.integer64(sample(c(-2L,0L,2L,NA), 20, TRUE)))
+nn = list(1, 5, -1, -5, c(1,2), c(-1,1))
+testnum = 2224
+for (n in nn) {
+  for (c in names(x)[-1]) {
+    testnum = testnum + 0.001
+    test(testnum, EVAL(sprintf("DT[, shift(%s, %d, type='lag'), by=x]", c, n)),  EVAL(sprintf("DT[, esc(%s, %d, type='lag'), by=x]", c, n)))
```

Comment:
I see that comparing gforce against non-gforce makes sense given all the other existing tests of non-gforce. But that could change in future (e.g. https://github.com/Rdatatable/data.table/pull/5205#discussion_r729489615). `y` should really be a fixed result so the tests are isolated and independent. I see the difficulty in constructing `y` given it's a nice loop through many cases. So we can save an `.Rdata` and compare to that. Will do. This will also also RAW to be easily tested here without having to add RAW to the regular shift.c which was the difficulty here I guess.